### PR TITLE
Feature/updating indeed scraper (#166)

### LINKED ISSUE #154: CAPTCHA has broken all scraping of Indeed
## Description

All users can scrape some jobs, but are quickly hit by captcha, and the scraping fails, yielding no jobs. 

Please go to #148 to discuss a resolution!

## Steps to Reproduce

1. Start a scrape job with default settings.

## Expected behavior

You get a hundred or so jobs.

## Actual behavior

You get nothing.

### PR DESCRIPTION
**Pull request by sammytheindi**
## Description

Merging fixed indeed scraper into master branch. For more information, see #166 
Additional changes include updates to formatting, adding black and prettier configuration files, and performing minor updates to the repo (versioning, updating modules, etc.). 

Biggest point of discussion is whether we want to keep the current method of scraping each individual job page, or scraping the job list with description summaries. The latter (default implementation in this PR), is **much** faster, though will result in less information in the description. It is possible to manually switch back, but do we want to keep it as the default implementation?

## Context of change

- [x] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Same tests as in #166 

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by PaulMcInnis — CHANGES_REQUESTED** [review]
lets put back the markdown + demo file but other than this i'm approving it

File: jobfunnel/resources/user_agent_list_mobile.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Mozilla/5.0 (iPhone; CPU iPhone OS 17_6_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 Indeed App 225.0
```
[the remark this answers is not in the document]
Comment (sammytheindi) [reply jobfunnel/resources/user_agent_list_mobile.txt:1]: We can actually get rid of the first one, only had it in as a reference. I assume that's what you mean?

**Review by PaulMcInnis — CHANGES_REQUESTED** [review]
wow makes me so happy to see this working again :) 

a few things

1. we should update readme to say that 3.11 is required at least
2. we need to add `user_agent_mobile.txt` to the `MANIFEST.in` file so that it is properly packaged when you install/run using `pip install .` and the `funnel load -s...` per the instructions for new users in the readme.

**Comment by sammytheindi** [comment]
Thanks @PaulMcInnis, we should be good to go! 

As I was updating, I was wondering why this project still uses `setup.py`, instead of the newer standard of `pyproject.toml` + `setup.cfg` ([PEP-518](https://peps.python.org/pep-0518/)). 

I will open a new issue on this as I believe we should probably switch while the project is still small, but would love to hear if there are any reasons not to.

**Comment by PaulMcInnis** [comment]
> Thanks @PaulMcInnis, we should be good to go!
> 
> As I was updating, I was wondering why this project still uses `setup.py`, instead of the newer standard of `pyproject.toml` + `setup.cfg` ([PEP-518](https://peps.python.org/pep-0518/)).


Hey there,  no reason other than age really - this is an old project and it needs some love

**Pull request merged**